src/cmakeTaskProvider.ts

```diff
@@ -162,6 +166,89 @@ export class CMakeTaskProvider implements vscode.TaskProvider {
         }
         return undefined;
     }
+
+    public static async resolveInternalTask(task: CMakeTask): Promise<CMakeTask | undefined> {
+        const execution: any = task.execution;
+        if (!execution) {
+            const definition: CMakeTaskDefinition = <any>task.definition;
+            const scope: vscode.WorkspaceFolder | vscode.TaskScope = vscode.TaskScope.Workspace;
+            const resolvedTask: CMakeTask = new vscode.Task(definition, scope, definition.label, CMakeTaskProvider.CMakeSourceStr,
+                new vscode.CustomExecution(async (resolvedDefinition: vscode.TaskDefinition): Promise<vscode.Pseudoterminal> =>
+                    new CustomBuildTaskTerminal(resolvedDefinition.command, resolvedDefinition.targets, resolvedDefinition.preset, resolvedDefinition.options)
+                ), []);
+            return resolvedTask;
+        }
+        return undefined;
+    }
+
+    public static async findBuildTask(presetName?: string, targets?: string[]): Promise<CMakeTask | undefined> {
+        // Fetch all CMake task from `tasks.json` files as well as tasks from task provider.
+        const allTasks: vscode.Task[] = await vscode.tasks.fetchTasks({ type: CMakeTaskProvider.CMakeScriptType });
+        const tasks: (CMakeTask | undefined)[] = allTasks.map((task: any) => {
+            if (!task.definition.label || !task.group || (task.group && task.group !== vscode.TaskGroup.Build)) {
```

Comment:
From what I have managed to find you cannot compare task.group with vscode.TaskGroup.Build
You need to compare "id" fields from these two.
This is because task.group is not exactly the same object.

In my case it leads to situation that task defined in tasks.json is skipped,
Group of tasks from tasks.json is:
![image](https://user-images.githubusercontent.com/29480839/205305609-181687ec-bce7-49c7-ad36-ff50b25aa534.png)

when vscode.TaskGroup.Build is:
![image](https://user-images.githubusercontent.com/29480839/205305730-10bb02e0-e6a0-42a8-8c1a-0a74a168f7be.png)